### ISSUE TITLE
Common Lisp lexer can't recognize ::symbol-name

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#518]
(Original issue 811 created by emosenkis on 2012-10-15T23:20:38.578273+00:00)

Because the 'symbol' regexp defined on line 736 of functional.py is expecting a complete symbol, it fails to recognize symbols that start ::, :#, and :. (those listed in the 'constituent' variable).
Based on testing in SBCL, all three are understood by the Lisp parser as symbols.

**Comment by Anteru** [comment]
(Original issue 811 last updated on 2013-05-06T06:27:21.775757+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by tshatch on 2012-11-06T16:18:16.030705+00:00)

I don't quite grok the problem, can you provide a complete statement that doesn't work currently, that I can use to test against?

**Comment by Anteru** [comment]
(Original comment by emosenkis on 2013-02-07T03:20:57.034074+00:00)

Sorry for the delay. The notification email accidentally got miscategorized and archived. A trivial example is (with the output when I typed them into sbcl):

```
#!common lisp

(keywordp ::junk)
  T
(symbol-name ::junk)
  "JUNK"

(symbol-name :#junk)
  "#JUNK"

(symbol-name #.junk)
  "#.JUNK"

```
These last two symbols fail to be recognized because there is no special case for symbols beginning with : - if you omit the colon, sbcl well likewise raise an error, but with the colon they are valid. The first case, with two colons, is a weird, but valid use of keywords. All symbols in the keyword package are automatically exported, making :kw, which is really a shortcut for keyword:kw, valid. However, it can be accessed without expecting it to be exported, as keyword::kw, and therefore as ::kw.

The Common Lisp lexer needs to be updated to properly lex these weird keywords.

**Comment by Anteru** [comment]
(Original comment by emosenkis on 2013-02-20T19:03:17.560485+00:00)

There appears to be a missing : in my last example. it should be:

```
#!common lisp

(symbol-name :#.junk)
  "#.JUNK"
```

As you can see, BitBucket isn't lexing it properly.

**Comment by Anteru** [comment]
(Original comment by tshatch on 2013-05-06T06:27:21.055818+00:00)

Added symbol handling for symbols that begin with :# and ::. Fixes #811

→ <<cset 693ed2e89835>>

**Comment by Anteru** [comment]
(Original comment by tshatch on 2013-05-06T06:27:21.769816+00:00)

Merged in russell/pygments-811 (pull request #189)

Added Commol Lisp symbol handling for symbols that begin with :# and ::. Fixes #811

→ <<cset ac9e4f8aefe9>>
